src/ProjectService.Business/Commands/Project/CreateProjectCommand.cs

```diff
@@ -34,36 +31,75 @@ public class CreateProjectCommand : ICreateProjectCommand
         private readonly IAccessValidator _accessValidator;
         private readonly ICreateProjectValidator _validator;
         private readonly ILogger<CreateProjectCommand> _logger;
-        private readonly IProjectInfoMapper _projectInfoMapper;
         private readonly IHttpContextAccessor _httpContextAccessor;
-        private readonly IRequestClient<IGetDepartmentRequest> _rcGetDepartment;
         private readonly IRequestClient<ICreateWorkspaceRequest> _rcCreateWorkspace;
+        private readonly IRequestClient<ICheckUsersExistence> _rcCheckUsersExistence;
         private readonly IRequestClient<ICreateWorkTimeRequest> _rcCreateWorkTime;
+        private readonly IRequestClient<ICheckDepartmentsExistence> _rcCheckDepartmentsExistence;
 
-        private IGetDepartmentResponse GetDepartment(Guid departmentId, List<string> errors)
+        private List<Guid> CheckDepartmentExistence(Guid? departmentId, List<string> errors)
         {
-            string errorMessage = "Cannot add project. Please try again later.";
+            if (!departmentId.HasValue)
+            {
+                return null;
+            }
+
+            string errorMessage = "Failed to check the existing department.";
```

Comment:
errorMessage nowhere used except before return null. Do not set errorMsg to a variable - set it to args